### PR TITLE
Add random key pair generation

### LINKED ISSUE #1408: Add method to generate key pair
### Feature Request

Add a method to generate a pair of keys in order to simplify creating an account. It can be potentially added as a static method `KeyPair.generate()` or `KeyPair.create()`.

### PR DESCRIPTION
**Pull request by franciszekjob**
Closes #1408

## Introduced changes

- Add `KeyPair.generate()`
- Move `KeyPair` into separate file
- Add tests for key pair generation

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: docs/account_creation.rst [hunk 1]
```diff
@@ -16,7 +16,7 @@ and then creating the transaction which will charge the fee from the address.
 Deploying an account with DeployAccount transaction requires the following:
 
  - class_hash of the account contract
- - generating a private key and deployment salt
+ - private key and deployment salt
```
Comment (franciszekjob) [inline docs/account_creation.rst:19]: note: Just to be precise, deploying an account requires  private key, not private key generation.

File: starknet_py/net/signer/key_pair.py
Comment (franciszekjob) [inline starknet_py/net/signer/key_pair.py:1]: note: `KeyPair` has been moved to separate file

File: starknet_py/net/signer/test_key_pair.py
Comment (franciszekjob) [inline starknet_py/net/signer/test_key_pair.py:1]: I've added this basic test for key pair generation. Actually, it would be good to provide tests for other methods, however this is not in the scope of this PR. Let's address it in separate one.

File: starknet_py/tests/e2e/docs/account_creation/test_deploy_prefunded_account.py [hunk 2]
```diff
@@ -29,6 +29,9 @@ async def test_deploy_prefunded_account(
 
     key_pair = KeyPair.from_private_key(private_key)
 
+    # You can also generate a key pair
+    key_pair_generated = KeyPair.generate()
+
```
Comment (franciszekjob) [inline starknet_py/tests/e2e/docs/account_creation/test_deploy_prefunded_account.py:34]: note: We could private separate section about different ways of key pair generation/reading, it would also cover `KeyPair.from_keystore()`. Can be addresses in separate PR as well.

File: starknet_py/net/signer/test_key_pair.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from starknet_py.constants import FIELD_PRIME
+from starknet_py.net.signer.key_pair import KeyPair
+
+
+def test_generate():
```
Comment (kkawula) [inline starknet_py/net/signer/test_key_pair.py:5]:
``` suggestion
def test_generate_key_pair():
```
nit: Consider renaming this test for more clarity.

File: starknet_py/tests/e2e/docs/account_creation/test_deploy_prefunded_account.py (hunk 2 above)
Comment (kkawula) [reply starknet_py/tests/e2e/docs/account_creation/test_deploy_prefunded_account.py:34]: It sounds good and will be helpful for new users, let's create an issue for that

File: starknet_py/net/signer/test_key_pair.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    key_pair = KeyPair.generate()
+    hex_private_key = hex(key_pair.private_key)[2:]
+    padded_hex_private_key = hex_private_key.zfill(64)
+    assert len(padded_hex_private_key) == 64
+    assert key_pair.private_key < FIELD_PRIME
+    assert key_pair.public_key < FIELD_PRIME
```
Comment (kkawula) [inline starknet_py/net/signer/test_key_pair.py:11]: I suggest to move this file to `test` directory with the spirit of the second paragraph of test refactoring issue #1219

File: starknet_py/tests/e2e/docs/account_creation/test_deploy_prefunded_account.py (hunk 2 above)
Comment (franciszekjob) [reply starknet_py/tests/e2e/docs/account_creation/test_deploy_prefunded_account.py:34]: Mentioned it in https://github.com/software-mansion/starknet.py/issues/1411.

**Review by kkawula — APPROVED** [review]
